db/db_impl/db_impl.cc

```diff
@@ -387,13 +387,19 @@ Status DBImpl::ResumeImpl(DBRecoverContext context) {
     }
   }
 
-  // We cannot guarantee consistency of the WAL. So force flush Memtables of
-  // all the column families
   if (s.ok()) {
-    FlushOptions flush_opts;
-    // We allow flush to stall write since we are trying to resume from error.
-    flush_opts.allow_write_stall = true;
-    s = FlushAllColumnFamilies(flush_opts, context.flush_reason);
+    if (context.flush_reason == FlushReason::kErrorRecoveryRetryFlush) {
+      s = RetryFlushesForErrorRecovery(FlushReason::kErrorRecoveryRetryFlush,
+                                       true /* wait */);
+    } else {
+      assert(context.flush_reason == FlushReason::kErrorRecovery);
```

Comment:
Can flush_reason not set at this point? 

Wondering if `context` is guaranteed to have `flush_reason` at the following line

https://github.com/facebook/rocksdb/blob/6b4315ee8badc2fd4dea2cade6cdd1acfb92db78/db/error_handler.cc#L702

Reply:
Not sure. I can delete this assertion - it should behave the same as before by calling FlushAllColumnFamilies() in any scenarios we don't know about.